src/view/reviewCommentController.ts

```diff
@@ -706,55 +660,28 @@ export class ReviewDocumentCommentProvider implements vscode.Disposable, Comment
 	// #endregion
 
 	// #region Review
-	public async startReview(thread: vscode.CommentThread): Promise<void> {
+	public async startReview(thread: GHPRCommentThread, input: string): Promise<void> {
 		await this._prManager.startReview(this._prManager.activePullRequest!);
-
-		if (thread.comments.length) {
-			let comment = thread.comments[0] as (vscode.Comment & { _rawComment: Comment });
-			const rawComment = await this._prManager.createCommentReply(this._prManager.activePullRequest!, this.commentController!.inputBox ? this.commentController!.inputBox!.value : '', comment._rawComment);
-
-			const vscodeComment = convertToVSCodeComment(rawComment!, undefined);
-			updateCommentCommands(vscodeComment,this.commentController!, thread, this._prManager.activePullRequest!, this);
-			thread.comments = [...thread.comments, vscodeComment];
-			updateCommentThreadLabel(thread);
-		} else {
-			// create new comment thread
-
-			if (this.commentController!.inputBox && this.commentController!.inputBox!.value) {
-				await this.updateCommentThreadRoot(thread, this.commentController!.inputBox!.value);
-			}
-		}
-
-		if (this.commentController!.inputBox) {
-			this.commentController!.inputBox!.value = '';
-		}
+		await this.createOrReplyComment(thread, input);
 	}
 
-	public async finishReview(thread: vscode.CommentThread): Promise<void> {
-		if (this.commentController && this.commentController.inputBox && this.commentController.inputBox.value) {
-			let comment = thread.comments[0] as (vscode.Comment & { _rawComment: Comment });
-			const rawComment = await this._prManager.createCommentReply(this._prManager.activePullRequest!, this.commentController!.inputBox!.value, comment._rawComment);
-			const vscodeComment = convertToVSCodeComment(rawComment!, undefined);
-			updateCommentCommands(vscodeComment, this.commentController!, thread, this._prManager.activePullRequest!, this);
-			thread.comments = [...thread.comments, vscodeComment];
-			updateCommentThreadLabel(thread);
-			this.commentController!.inputBox!.value = '';
+	public async finishReview(thread: GHPRCommentThread, input: string): Promise<void> {
+		try {
+			this.createOrReplyComment(thread, input);
```

Comment:
Should `createOrReplyComment()` be awaited here?

Reply:
yeah, you're right